services/core/events/blockchain.js

```diff
@@ -30,31 +30,34 @@ module.exports = [
 		description: 'Keep the block list up-to-date',
 		controller: callback => {
 			Signals.get('newBlock').add(async data => {
-				logger.debug(`New block arrived (${data.id})...`);
-
-				// Fork detection
-				if (localPreviousBlockId) {
-					if (localPreviousBlockId !== data.previousBlockId) {
-						logger.debug(`Fork detected at block height ${localPreviousBlockId}`);
+				if (data && Array.isArray(data.data)) {
```

Comment:
Is it the case here that data is undefined? I remember that happened before but not sure if it still does work that way. Can you please elaborate on that?

Reply:
It works perfectly as of now. I had asked to add the checks as a measure of safety. Perhaps we should also consider adding a `WARN` logger in the `else` section to highlight the failures.